spec/acceptance/letsencrypt_plugin_dns_cloudflare_spec.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'spec_helper_acceptance'
+
+describe 'letsencrypt::plugin::dns_cloudflare' do
+  it_behaves_like 'an idempotent resource' do
+    let(:manifest) do
+      <<-PUPPET
+      class { 'letsencrypt' :
+        email  => 'letsregister@example.com',
+        config => {
+          'server' => 'https://acme-staging-v02.api.letsencrypt.org/directory',
+        },
+      }
+      class { 'letsencrypt::plugin::dns_cloudflare':
+        api_token => 'dummy-cloudflare-api-token',
+      }
+      Class['letsencrypt'] -> Class['letsencrypt::plugin::dns_cloudflare']
```

Comment:
I would feel better if this would be part of the class itself, not in the acceptance test. Otherwise it needs to be repeated for every use of it. Remember that `require $class` is the same as `include $class ; Class[$class] -> Class[$self]`.

Reply:
Yes, but the class `letsencrypt::plugin::dns_cloudflare` should never be included if you use the `letsencrypt` plugin... it is enough to just use

```
include letsencrypt

letsencrypt::certonly { 'example.org':
  plugin => 'dns-cloudflare',
}
```

because the `letsencrypt::plugin::dns_cloudflare` class will be included then automatically. So a call like in the acceptance test is not required to use this plugin. If i add `require letsencrypt` into `letsencrypt::plugin::dns:_cloudflare` it will end into a dependency cycle at least if i use this module with puppet bolt. This was why i removed the require inside the plugin itself.

Reply:
As suggested in https://github.com/voxpupuli/puppet-letsencrypt/pull/279#discussion_r855025446, `require letsencrypt::install` might be sufficient then.

> If i add `require letsencrypt` into `letsencrypt::plugin::dns:_cloudflare` it will end into a dependency cycle at least if i use this module with puppet bolt. This was why i removed the require inside the plugin itself.

The spec test you added should also cover this.

Reply:
Okay ... changed it. Worked also with bolt in one of my projects. 👍